RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -120,16 +167,50 @@ void Phase2TrackerRecHitsValidation::analyze(const edm::Event& event, const edm:
     const TrackerTopology* tTopo = tTopoHandle.product();
 
     /*
+     * Rearrange the simTracks
+     */
+
+    // Rearrange the simTracks for ease of use <simTrackID, simTrack>
+    SimTracksMap simTracks;
+    for (edm::SimTrackContainer::const_iterator simTrackIt(simTracksRaw->begin()); simTrackIt != simTracksRaw->end(); ++simTrackIt) simTracks.insert(std::pair< unsigned int, SimTrack >(simTrackIt->trackId(), *simTrackIt));
+
+    /*
+     * Rearrange the simHits by detUnit
+     */
+
+    // Rearrange the simHits for ease of use 
+    SimHitsMap simHitsDetUnit;
+    SimHitsMap simHitsTrackId;
+    for (unsigned int simhitidx = 0; simhitidx < 2; ++simhitidx) {
+      for (edm::PSimHitContainer::const_iterator simHitIt(simHitsRaw[simhitidx]->begin()); simHitIt != simHitsRaw[simhitidx]->end(); ++simHitIt) {
+        SimHitsMap::iterator simHitsDetUnitIt(simHitsDetUnit.find(simHitIt->detUnitId()));
+        if (simHitsDetUnitIt == simHitsDetUnit.end()) {
+            std::pair< SimHitsMap::iterator, bool > newIt(simHitsDetUnit.insert(std::pair< unsigned int, std::vector< PSimHit > >(simHitIt->detUnitId(), std::vector< PSimHit >())));
+            simHitsDetUnitIt = newIt.first;
+        }
+        simHitsDetUnitIt->second.push_back(*simHitIt);
+
+        SimHitsMap::iterator simHitsTrackIdIt(simHitsTrackId.find(simHitIt->trackId()));
+        if (simHitsTrackIdIt == simHitsTrackId.end()) {
+            std::pair< SimHitsMap::iterator, bool > newIt(simHitsTrackId.insert(std::pair< unsigned int, std::vector< PSimHit > >(simHitIt->trackId(), std::vector< PSimHit >())));
+            simHitsTrackIdIt = newIt.first;
+        }
+        simHitsTrackIdIt->second.push_back(*simHitIt);
+      }
+    }
+
+    /*
      * Validation   
      */
 
+
     // Loop over modules
     for (Phase2TrackerRecHit1DCollectionNew::const_iterator DSViter = rechits->begin(); DSViter != rechits->end(); ++DSViter) {
 
         // Get the detector unit's id
         unsigned int rawid(DSViter->detId()); 
         DetId detId(rawid);
-        unsigned int layer(getLayerNumber(detId, tTopo));
+        unsigned int layer = tTopo->side(detId)*100 + tTopo->layer(detId);
```

Comment:
I changed the only one that is under my responsibility, namely RecoLocalTracker/SiPhase2Clusterizer/test/ClustersValidation.cc
I added it to this PR now.